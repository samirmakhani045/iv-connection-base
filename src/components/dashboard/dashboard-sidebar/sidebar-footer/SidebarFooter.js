import React from 'react';
import PropTypes from 'prop-types';

import { StyledButton, Title, P } from './SidebarFooter.styles';

const SidebarFooter = (props) => {
  const { title, text, button } = props;

  return (
    <div data-testid="wrapper">
      {title && <Title>{title}</Title>}
      {text && <P>{text}</P>}
      <StyledButton variant="light" {...button} />
    </div>
  );
};

SidebarFooter.propTypes = {
  title: PropTypes.string,
  text: PropTypes.string,
  button: PropTypes.shape({
    label: PropTypes.string,
    icon: PropTypes.string,
    logo: PropTypes.string,
    onClick: PropTypes.func,
  }),
};

export default SidebarFooter;
