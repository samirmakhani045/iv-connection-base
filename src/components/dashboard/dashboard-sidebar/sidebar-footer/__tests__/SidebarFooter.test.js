import React from 'react';
import { render, screen } from '@testing-library/react';

import SidebarFooter from '../SidebarFooter';

const defaultProps = {
  title: 'Title',
  text: 'Text',
  button: {
    label: 'Button',
    icon: 'Add',
  },
};

const Test = (props) => <SidebarFooter {...defaultProps} {...props} />;

describe('<SidebarFooter />', () => {
  it('should render correctly', () => {
    const { container } = render(<Test />);

    expect(screen.queryByTestId('wrapper')).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
