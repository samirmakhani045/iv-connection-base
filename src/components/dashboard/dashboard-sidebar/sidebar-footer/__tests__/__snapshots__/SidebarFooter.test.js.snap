// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SidebarFooter /> should render correctly 1`] = `
<div
  data-testid="wrapper"
>
  <p
    class="SidebarFooterstyles__Title-sc-7jmd2j-0 fWRSdH"
  >
    Title
  </p>
  <p
    class="SidebarFooterstyles__P-sc-7jmd2j-1 dJkCpp"
  >
    Text
  </p>
  <button
    class="Buttonstyles__StyledButton-sc-1uuuyg7-0 jPrzzZ SidebarFooterstyles__StyledButton-sc-7jmd2j-2 nVHiT"
    data-testid="button"
    type="button"
  >
    <div
      class="Iconstyles__IconWrapper-sc-1ghvgbo-0 ipHglB Buttonstyles__StyledIcon-sc-1uuuyg7-1 fWCKUg"
      data-testid="buttonIcon"
      size="16"
    >
      <svg
        data-testid="icon"
        fill="#313336"
        height="1em"
        viewBox="0 0 24 24"
        width="1em"
      >
        <path
          d="M19 13H13V19H11V13H5V11H11V5H13V11H19V13Z"
        />
      </svg>
    </div>
    <span>
      Button
    </span>
  </button>
</div>
`;
